src/main/java/it/trade/api/TradeItApiClient.java

```diff
@@ -72,19 +75,10 @@ private void injectSession(TradeItRequestWithSession request) {
 
     public void getAvailableBrokers(final TradeItCallback<List<Broker>> callback) {
         TradeItRequestWithKey request = new TradeItRequestWithKey();
-        tradeItApi.getAvailableBrokers(request).enqueue(new Callback<TradeItAvailableBrokersResponse>() {
-            List<Broker> brokerList = new ArrayList<>();
-
-            public void onResponse(Call<TradeItAvailableBrokersResponse> call, Response<TradeItAvailableBrokersResponse> response) {
-                if (response.isSuccessful() && response.body().status == TradeItResponseStatus.SUCCESS) {
-                    TradeItAvailableBrokersResponse availableBrokersResponse = response.body();
-                    brokerList = availableBrokersResponse.brokerList;
-                }
-                callback.onSuccess(brokerList);
-            }
-
-            public void onFailure(Call call, Throwable t) {
-                callback.onSuccess(brokerList);
+        tradeItApi.getAvailableBrokers(request).enqueue(new DefaultCallbackWithErrorHandling<TradeItAvailableBrokersResponse, List<Broker>>(callback) {
```

Comment:
Yes. Because now we are caching the result, we need to return onError if it fails. Otherwise the user will have an empty list until he restarts the app.